lxd/identity/cache.go

```diff
@@ -15,7 +15,10 @@ type Cache struct {
 	// entries is a map of authentication method to map of identifier to CacheEntry. The identifier is either a
 	// certificate fingerprint (tls) or an email address (oidc).
 	entries map[string]map[string]*CacheEntry
-	mu      sync.RWMutex
+
+	// identityProviderGroups is a map of identity provider group name to slice of LXD group names.
+	identityProviderGroups map[string][]string
```

Comment:
Should we use `map[string]*[]string`  to avoid growing the cache permanently  unnecessarily?